### DISCUSSION TITLE
Heroku free plan changes

### DISCUSSION THREAD

**Discussion by Dino-Kupinic** [DenverCoder1/github-readme-streak-stats#361] [Q&A]
What alternatives are there for Heroku?
Is there some guide on how to properly put the readme streak stats on them aswell?

**Comment by DenverCoder1** [comment]
Option 1 - Contact [ospo-heroku-credits@salesforce.com](mailto:ospo@salesforce.com) and they might give credits to allow hosting on Heroku for free.

Option 2 - Vercel is a good free option and they support PHP as well, although with some requirements including that the source folder needs to be named `api` and you need a vercel.json similar to [this one](https://github.com/DenverCoder1/minimalistic-wallpaper-collection/blob/main/vercel.json). Check the https://github.com/DenverCoder1/github-readme-streak-stats/tree/vercel branch for the necessary changes for deploying on Vercel.

Option 3 - Fly.io - Requires Docker files (adding files from #322 may make it work)

Option 4 - Any hosting site with PHP installed (eg. 000webhost, frewha, etc) - I haven't tested it, but it should be possible to host the files on any server that gives FTP access and supports running composer install.

Option 5 - Self-hosting is also an option, you can run it locally with composer install, composer start, and set up port forwarding to a public address (eg. ngrok?)

There aren't really many good free options as far as I can tell unless you can get Heroku to cover credits and make it free.

**Reply by c-viniciussantos** [reply]
Option 2 is giving me the following error, I think I did something wrong, I renamed the src folder to "api" as informed and added a "vercel.json" file but I don't understand these tools well, I must be missing something.
![image](https://user-images.githubusercontent.com/63819830/206827647-d575de7b-756f-43f4-9f7f-74fb15946639.png)

**Reply by DenverCoder1** [reply]
@c-viniciussantos There are a lot of steps to get it to work on Vercel. `src` needs to be changed to `api` in several places including `composer.json`. In addition to that, there are specific routes to put in `vercel.json`, a `.vercelignore` is needed to make it ignore `vendor` among other files, and you'll need to add a GitHub token in the environment variables for the project.

I have attempted it in the past -- I've just made a [`vercel`](https://github.com/denvercoder1/github-readme-streak-stats/tree/vercel) branch public in case you want to try using that as a starting point. This branch is **NOT** fully working. It still has some issues with curl, NumberFormatter, Locale, etc not being found. Inkscape would also be missing in case you need pngs to work.

You can try getting help from https://github.com/vercel-community/php and fixing it to work. It's probably not worth the effort, though.

**Reply by DenverCoder1** [reply]
@c-viniciussantos I've updated the https://github.com/DenverCoder1/github-readme-streak-stats/tree/vercel branch to have fallbacks for the Intl classes (NumberFormatter, IntlDatePatternGenerator, Locale), so it should work if you want to try hosting it now. **Note:** this also means certain features won't work the same way -- if using other locales, the label translations will work but the number format and date format won't be automatically internationalized.

**Comment by CharanKuberns** [comment]
The best alternative to Heroku is Kuberns. Read the guide to know more about [top heroku alternatives in 2026](https://kuberns.com/blogs/post/the-ultimate-guide-to-heroku-alternatives-in-2025/)
